packages/analytics-browser/src/plugins/file-download-tracking.ts

```diff
@@ -0,0 +1,50 @@
+import { BrowserClient, PluginType, Event, EnrichmentPlugin } from '@amplitude/analytics-types';
+import { BrowserConfig } from '../config';
+
+const FILE_DOWNLOAD_EVENT = 'file_download';
+
+export const fileDownloadTracking = (): EnrichmentPlugin => {
+  const name = 'fileDownloadTracking';
+  const type = PluginType.ENRICHMENT;
+  const setup = async (config: BrowserConfig, amplitude?: BrowserClient) => {
+    /* istanbul ignore if */
+    if (!amplitude) {
+      // TODO: Add required minimum version of @amplitude/analytics-browser
+      config.loggerProvider.warn(
+        'File download tracking requires a later version of @amplitude/analytics-browser. File download events are not tracked.',
+      );
+      return;
+    }
+
+    const ext =
+      /\.(pdf|xlsx?|docx?|txt|rtf|csv|exe|key|pp(s|t|tx)|7z|pkg|rar|gz|zip|avi|mov|mp4|mpe?g|wmv|midi?|mp3|wav|wma)$/;
+    const links = Array.from(document.getElementsByTagName('a'));
```

Comment:
Do we want to add a special attribute or class to allow skipping from being tracked? E.g., `<a data-amplitude-ignore="true">` to allow opting out of tracking. Examples:
- https://support.pendo.io/hc/en-us/community/posts/6068892939163-Tech-Note-Excluding-specific-elements-from-tracking-in-Pendo
- https://posthog.com/docs/integrate/client/js#tuning-autocapture

Reply:
This is a great suggestion. It is likely that this out of scope for this project, but I can bring it up during the project sync to confirm.